modules/sources/transformers/swap-v2-transformer.ts

```diff
@@ -14,10 +14,40 @@ export function swapV2Transformer(swap: BalancerSwapFragment, chain: Chain): Swa
     // Avoiding scientific notation
     const feeFloat = parseFloat(swap.tokenAmountIn) * parseFloat(swap.poolId.swapFee ?? 0);
     const fee = feeFloat < 1e6 ? feeFloat.toFixed(18).replace(/0+$/, '').replace(/\.$/, '') : String(feeFloat);
-    const feeFloatUSD = parseFloat(swap.valueUSD) * parseFloat(swap.poolId.swapFee ?? 0);
-    const feeUSD =
+    let feeFloatUSD = parseFloat(swap.valueUSD) * parseFloat(swap.poolId.swapFee ?? 0);
+    let feeUSD =
         feeFloatUSD < 1e6 ? feeFloatUSD.toFixed(18).replace(/0+$/, '').replace(/\.$/, '') : String(feeFloatUSD);
 
+    // FX pools have a different fee calculation
+    // Replica of the subgraph logic:
+    // https://github.com/balancer/balancer-subgraph-v2/blob/60453224453bd07a0a3a22a8ad6cc26e65fd809f/src/mappings/vault.ts#L551-L564
+    if (swap.poolId.poolType === 'FX') {
+        const USDC_ADDRESS = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48';
```

Comment:
How can we hardcode this? Isnt this different for each chain?

Reply:
Do we only compare against USDC? No other FOREX such as EURC?

Reply:
It looks like all the markets are against USDC, but true that it needs to be done for other chains as well.

Reply:
Hi @franzns, @gmbronco, would it be possible to add a `quoteToken` property on the Pool model? This new prop will hold the quote token address, whether USDC, USDC.e, or any token.

This is how we assign the `quoteToken` value in our balancer-subgraph fork:
```
function handleNewFXPool(event: ethereum.Event, permissionless: boolean): void {
...
  pool.quoteToken = getFXPoolQuoteToken(poolAddress);
...
}

function getFXPoolQuoteToken(poolAddress: Address): Bytes {
  let poolContract = FXPool.bind(poolAddress);
  // FXPool **always** holds the quote token in the second position
  let call = poolContract.try_derivatives(BigInt.fromI32(1));
  if (call.reverted) {
    log.error('Failed to get quote token for FXPool: {}', [poolAddress.toHexString()]);
    return stringToBytes('0x');
  }
  return call.value;
}
```

We then refer to `quoteToken` instead of hardcoded addresses to determine base and quote:
```
let isTokenInBase = tokenOutAddress === pool.quoteToken;
```

Reply:
> Hi @franzns, @gmbronco, would it be possible to add a `quoteToken` property on the Pool model? This new prop will hold the quote token address, whether USDC, USDC.e, or any token.
> 
> This is how we assign the `quoteToken` value in our balancer-subgraph fork:
> 
> ```
> function handleNewFXPool(event: ethereum.Event, permissionless: boolean): void {
> ...
>   pool.quoteToken = getFXPoolQuoteToken(poolAddress);
> ...
> }
> 
> function getFXPoolQuoteToken(poolAddress: Address): Bytes {
>   let poolContract = FXPool.bind(poolAddress);
>   // FXPool **always** holds the quote token in the second position
>   let call = poolContract.try_derivatives(BigInt.fromI32(1));
>   if (call.reverted) {
>     log.error('Failed to get quote token for FXPool: {}', [poolAddress.toHexString()]);
>     return stringToBytes('0x');
>   }
>   return call.value;
> }
> ```
> 
> We then refer to `quoteToken` instead of hardcoded USDC addresses to determine base and quote:
> 
> ```
> let isTokenInBase = tokenOutAddress === pool.quoteToken;
> ```

Yes, we should definitely do that. We already have pool type specific data where we can add the quote token. Is the quote token immutable?

Reply:
Yes, the quote token is immutable for FX pools